linkml_runtime/utils/schemaview.py

```diff
@@ -529,6 +531,40 @@ def expand_curie(self, uri: str) -> str:
                     return ns[pfx] + local_id
         return uri
 
+    @lru_cache(CACHE_SIZE)
+    def get_element_by_prefix(
```

Comment:
perhaps decompose this:

 - a method that takes an actual prefix as input and returns element names, `get_elements_applicable_for_prefix`
 - a method that splits CURIEs (or URIs?) and. extracts the prefix
 - a wrapper that composes these `get_elements_applicable_for_identifier`

Reply:
I went ahead and broke this up into 3 methods and added support (by utilizing an existing function! - thanks for the pointer :) ) for URIs as well.